contracts/schema/Schema.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.7.0 <0.9.0;
+/**
+ * @title Schema
+ * @dev Schema contract
+ */
+contract SchemaRegistry {
+
+    struct Schema {     
+        address issuer;
+        string name;
+        bytes32 hash;  
+        bytes body;
+        uint256 timestamp;
+    }
+
+    mapping(string => bytes32) nameHash;
+    mapping(bytes32 => Schema) hashSchema;
+    
+    function getHashFromBytes(bytes memory schemaBody) private pure returns (bytes32) {
+        return keccak256(schemaBody);
```

Comment:
May be there is no reason to has inside the contract but do it outside in some service

Reply:
we understand that keccak256 is quite costly, but even if the hash is passed from outside as a parameter we need to verify it inside the contract. Now with @demonsh we agreed to calculate hash here.